gobblin-api/src/main/java/org/apache/gobblin/password/PasswordManager.java

```diff
@@ -64,44 +72,77 @@
   private static final long CACHE_SIZE = 100;
   private static final long CACHE_EXPIRATION_MIN = 10;
   private static final Pattern PASSWORD_PATTERN = Pattern.compile("ENC\\((.*)\\)");
+  private final boolean useStrongEncryptor;
+  private FileSystem fs;
+  private List<String> masterPasswords;
 
-  private static final LoadingCache<Map.Entry<Optional<String>, Boolean>, PasswordManager> CACHED_INSTANCES =
+  private static final LoadingCache<CachedInstanceKey, PasswordManager> CACHED_INSTANCES =
       CacheBuilder.newBuilder().maximumSize(CACHE_SIZE).expireAfterAccess(CACHE_EXPIRATION_MIN, TimeUnit.MINUTES)
-          .build(new CacheLoader<Map.Entry<Optional<String>, Boolean>, PasswordManager>() {
-
+          .build(new CacheLoader<CachedInstanceKey, PasswordManager>() {
             @Override
-            public PasswordManager load(Map.Entry<Optional<String>, Boolean> cacheKey) {
-              return new PasswordManager(cacheKey.getKey(), cacheKey.getValue());
+            public PasswordManager load(CachedInstanceKey cacheKey) {
+              return new PasswordManager(cacheKey);
             }
           });
 
-  private Optional<TextEncryptor> encryptor;
+  private Optional<TextEncryptor> encryptor = Optional.absent();
 
-  private PasswordManager(Optional<String> masterPassword, boolean useStrongEncryptor) {
-    if (masterPassword.isPresent()) {
-      this.encryptor = useStrongEncryptor ? Optional.of((TextEncryptor) new StrongTextEncryptor())
-          : Optional.of((TextEncryptor) new BasicTextEncryptor());
-      try {
+  private PasswordManager(CachedInstanceKey cacheKey) {
+    this.useStrongEncryptor = cacheKey.useStrongEncryptor;
 
-        // setPassword() needs to be called via reflection since the TextEncryptor interface doesn't have this method.
-        this.encryptor.get().getClass().getMethod("setPassword", String.class).invoke(this.encryptor.get(),
-            masterPassword.get());
-      } catch (Exception e) {
-        LOG.error("Failed to set master password for encryptor", e);
-        this.encryptor = Optional.absent();
+    try {
+      this.fs = cacheKey.fsURI != null ? FileSystem.get(URI.create(cacheKey.fsURI), new Configuration())
+          : (cacheKey.masterPasswordFile != null ? new Path(cacheKey.masterPasswordFile).getFileSystem(new Configuration()) : null);
+    } catch (IOException e) {
+      LOG.warn("Failed to instantiate FileSystem.", e);
+    }
+    this.masterPasswords = getMasterPasswords(cacheKey);
+  }
+
+  private List<String> getMasterPasswords(CachedInstanceKey cacheKey) {
+    List<String> masterPasswordFiles = new ArrayList<>();
```

Comment:
masterPasswordFiles -> masterPasswords since this is a list of passwords.

Reply:
done